lineage/CellVar.py

```diff
@@ -0,0 +1,106 @@
+# Variables with the prefix num (i.e., num_states, num_cells) have an underscore following the 'num'.
+# Docstrings use """ and not '''.
+# Class names use camelCase.
+
+
+class CellVar:
+    def __init__(self, state, left, right, parent, gen):
+        """ Instantiates the cell object. Contains memeber variables that identify daughter cells and parent cells. Also contains the state of the cell. """
+        self.state = state
+        self.left = left
+        self.right = right
+        self.parent = parent
+        self.gen = gen
+
+    def _divide(self, T):
+        """ Member function that performs division of a cell. Equivalent to adding another timestep in a Markov process. """
+        left_state, right_state = _double(self.state, T)  # roll a loaded die according to the row in the transtion matrix
+        self.left = CellVar(state=left_state, left=None, right=None, parent=self, gen=self.gen + 1)  # assign the resulting states to new cells
+        self.right = CellVar(state=right_state, left=None, right=None, parent=self, gen=self.gen + 1)  # ensure that those cells are related
+
+        return self.left, self.right
+
+    def _isParent(self):
+        """ Boolean. Returns true if the cell has daughters. """
+        return self.left is not None or self.right is not None
+
+    def _isChild(self):
+        """ Boolean. Returns true if this cell has a known parent. """
+        if self.parent:
+            return self.parent._isParent()
+        else:
+            return False
+
+    def _isRootParent(self):
+        """ Boolean. Returns true if this cell is the first cell in a lineage. """
+        bool_parent = False
+        if not self.parent and self.gen == 1:
+            bool_parent = True
+        return bool_parent
+
+    def _isLeaf(self):
+        """ Boolean. Returns true when a cell is a leaf with no children. """
+        return self.left is None and self.right is None
+
+    def _get_sister(self):
+        """ Member function that gets the sister of the current cell. """
+        cell_to_return = None
+        if self.parent.left is self:
+            cell_to_return = self.parent.right
+        elif self.parent.right is self:
+            cell_to_return = self.parent.left
+        return cell_to_return
+
+    def _get_root_cell(self):
+        """ Get the first cell in the lineage to which this cell belongs. """
+        curr_cell = self
+        while curr_cell.gen > 1:
+            curr_cell = curr_cell.parent
+        assert curr_cell._isRootParent()
+        return curr_cell
+    
+    def _get_daughters(self):
+        """ Get the left and right daughters of a cell. """
+        temp = []
+        if self.left is not None:
+            temp.append(self.left)
+        if self.right is not None:
+            temp.append(self.right)
+        return temp
+
+    def __repr__(self):
+        if hasattr(self, 'obs'):
+            return "Generation: {}, State: {}, Observation: {}".format(self.gen, self.state, self.obs)
```

Comment:
what is self.obs ? where is it defined ?

Reply:
it doesn't need to be defined upon instantiation. All cells are empty (without observations) in the beginning. We then fill their observations in when the state distributions are known. This is done in LineageTree.py.